litr/src/test/java/com/linkedin/android/litr/render/AudioRendererShould.kt

```diff
@@ -0,0 +1,187 @@
+package com.linkedin.android.litr.render
+
+import android.media.MediaCodec
+import android.media.MediaFormat
+import com.linkedin.android.litr.codec.Encoder
+import com.linkedin.android.litr.codec.Frame
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import java.nio.ByteBuffer
+import java.util.concurrent.CountDownLatch
+import kotlin.test.assertNotNull
+
+private const val INPUT_FRAME_TAG = 0
+private const val OUTPUT_FRAME_TAG_1 = 1
+private const val OUTPUT_FRAME_TAG_2 = 2
+private const val INPUT_BUFFER_SIZE = 42
+private const val PRESENTATION_TIME = 100L
+private const val ENCODER_BUFFER_CAPACITY = 42
+private const val EXTRA_INPUT_BYTES = 10
+private const val SOURCE_SAMPLE_RATE = 44100
+private const val TARGET_SAMPLE_RATE = 44100
+
+class AudioRendererShould {
+
+    @Mock private lateinit var encoder: Encoder
+
+    private val encoderInputFrame1 =
+        Frame(OUTPUT_FRAME_TAG_1, ByteBuffer.allocate(ENCODER_BUFFER_CAPACITY), MediaCodec.BufferInfo())
+
+    private val encoderInputFrame2 =
+        Frame(OUTPUT_FRAME_TAG_2, ByteBuffer.allocate(ENCODER_BUFFER_CAPACITY), MediaCodec.BufferInfo())
+
+    private lateinit var audioRenderer: AudioRenderer
+
+    private lateinit var encoderLatch: CountDownLatch
+    private val outputFrames = mutableListOf<Frame>()
```

Comment:
nit: you can potentially avoid having this state up here by moving this to a method that returns something like a future/callable on completion, e.g.
```kotlin
fun Encoder.collectOutputFrames(count: Int): Callable<List<Frame>> {
    val latch = CountDownLatch(count)
    val frames = mutableListOf<Frame>()
    whenever(this.queueInputFrame(any())).thenAnswer {
        frames.add(it.arguments[0] as Frame)
        latch.countDown()
    }
    return Callable {
        latch.await()
        frames
    }
}
```

Then you would do something like,
```kotlin
val collect = encoder.collectOutputFrames(2)
// ... act ...
val outputFrames = collect.call()
```

Reply:
Wow, this is a lot more elegant than what I have. Will change my code to use this.